### PR TITLE
operator: Add flag to forcefully update CRDs

### PR DESCRIPTION
**Pull request by aohoyd**
Currently tetragon-operator allows to update CRDs only if running version is smaller. But in some cases it is required to just replace CRDs.

This PR makes next changes:
- Adds `pkg/k8s/crdutil/CRDOption` struct and `pkg/k8s/crdutil/RegisterCRDsWithOptions` function to pass options into control flow
- Adds flag to the operator
- Adds flag to the helm chart

Fixes: #2014

```release-note
Allow for the operator to forcefully update CRDs regardless of the version
```

### CODE REVIEW

File: pkg/k8s/crdutils/register.go [hunk 1]
```diff
@@ -71,20 +76,30 @@ func NewCRD(crdName, resName string, crd apiextensionsv1.CustomResourceDefinitio
 func RegisterCRDs(clientset apiextensionsclient.Interface, crds []CRD) error {
 	g, _ := errgroup.WithContext(context.Background())
 	g.Go(func() error {
-		return createCRDs(clientset, crds)
+		return createCRDs(clientset, crds, CRDOptions{})
+	})
+
+	return g.Wait()
```
Comment (kkourt) [inline pkg/k8s/crdutils/register.go:82]: nit: how about just `return RegisterCRDsWithOptions( CRDOptions{})` here?

File: operator/cmd/common/common.go [hunk 2]
```diff
@@ -22,6 +22,7 @@ func AddCommonFlags(cmd *cobra.Command) {
 	flags.String(operatorOption.ConfigDir, "", "Directory in which tetragon-operator-config configmap is mounted")
 	flags.Bool(operatorOption.SkipPodInfoCRD, false, "When true, PodInfo Custom Resource Definition (CRD) will not be created")
 	flags.Bool(operatorOption.SkipTracingPolicyCRD, false, "When true, TracingPolicy and TracingPolicyNamespaced Custom Resource Definition (CRD) will not be created")
+	flags.Bool(operatorOption.ForceUpdateCRD, false, "When true, operator will ignore current CRD version and forcefully update it")
```
Comment (kkourt) [inline operator/cmd/common/common.go:25]: I think CRDs is more precise. That is:
> 	flags.Bool(operatorOption.ForceUpdateCRDs, false, "When true, operator will ignore the current CRD versions and forcefully update them")

File: operator/option/config.go [hunk 3]
```diff
@@ -30,6 +30,10 @@ const (
 	// SkipTracingPolicyCRD specifies whether the tracing-policies CustomResourceDefinition will be
 	// disabled
 	SkipTracingPolicyCRD = "skip-tracing-policy-crd"
+
+	// ForceUpdateCRD specifies whether operator should ignore current CRD version
+	// and forcefully update it.
+	ForceUpdateCRD = "force-update-crd"
 )
```
Comment (kkourt) [inline operator/option/config.go:37]: Again, plura makes more sense to mere here. That is, `force-update-crds`.

**Review by kkourt — COMMENTED** [review]
Thanks!

I have some minor nits, but this LGTM!

File: pkg/k8s/crdutils/register.go (hunk 1 above)
Comment (aohoyd) [reply pkg/k8s/crdutils/register.go]: It makes sense, thank you! I've replaced duplicated code

File: operator/cmd/common/common.go (hunk 2 above)
Comment (aohoyd) [reply operator/cmd/common/common.go]: Changed it in all places

File: operator/option/config.go (hunk 3 above)
Comment (aohoyd) [reply operator/option/config.go]: Updated in operator and chart

**Comment by aohoyd** [comment]
@kkourt Can you please take a look at my patches?

**Comment by kkourt** [comment]
> @kkourt Can you please take a look at my patches?

Sure!

Can you please squash the changes from my feedback into the relevant original commits? (`git rebase --interactive` using the `squash` and `fixup` actions should help)

Also, when done please force-push the new branch and notify me bu clicking on the github button "Re-request review".

**Review by kkourt — APPROVED** [review]
LGTM, thank you!

**Pull request merged**